pkgs/build-support/replace-vars/default.nix

```
@@ -0,0 +1,77 @@
+{ lib, stdenvNoCC }:
+
+/**
+  `replaceVars` is a wrapper around the [bash function `substitute`](https://nixos.org/manual/nixpkgs/stable/#fun-substitute)
+  in the stdenv. It allows for terse replacement of names in the specified path, while checking
+  for common mistakes such as naming a substitution that does nothing or forgetting a substitution.
+
+  As with the [`--subst-var-by`](https://nixos.org/manual/nixpkgs/stable/#fun-substitute-subst-var-by)
+  flag, names are encoded as `@name@` in the provided file at the provided path.
+
+  Any unmatched variable names in the file at the provided path will cause a build failure.
+
+  Any remaining text that matches `@[A-Za-z_][0-9A-Za-z_'-]@` in the output after replacement
+  has occurred will cause a build failure.
+
+  Attribute arguments:
+
+  - `path`: A [string-like value](https://nixos.org/manual/nixpkgs/stable/#function-library-lib.strings.isStringLike) of a [store path](https://nixos.org/manual/nix/stable/store/store-path.html#store-path).
+
+  - `attrs`: each entry in this set corresponds to a `--subst-var-by` entry in `substitute`.
+    See https://nixos.org/manual/nixpkgs/stable/#fun-substitute
+
+  Example:
```

```suggestion
  # Example
```

This isn't 100% conform with the doc conventions, but we're not rendering this file anyways, [for now](https://github.com/nix-community/nixdoc/issues/21).